packages/api/core/src/util/plugin-interface.ts

```diff
@@ -25,12 +25,12 @@ function isForgePlugin(plugin: IForgePlugin | unknown): plugin is IForgePlugin {
 }
 
 export default class PluginInterface implements IForgePluginInterface {
-  private plugins: IForgePlugin[];
+  private plugins: Promise<IForgePlugin>[];
```

Comment:
Dangling promises isn't a great system, this should probably be the same type and then have a `PluginInterface.load()` method of something that is async and awaits the promises